services/api/src/models/definitions/user.json

```diff
@@ -48,98 +48,93 @@
-      "writeScopes": "none"
+      "writeAccess": "none"
     },
-    "authInfo": [
-      {
-        "lastUsedAt": {
-          "type": "Date",
-          "required": true,
-          "writeScopes": "none"
+    "$private": {
+      "type": "Scope",
+      "readAccess": "none",
+      "writeAccess": "none",
+      "attributes": {
+        "hashedPassword": {
+          "type": "String"
         },
-        "userAgent": {
-          "type": "String",
-          "required": false,
-          "writeScopes": "none"
-        },
-        "ip": {
-          "type": "String",
-          "required": true,
-          "writeScopes": "none"
+        "loginAttempts": {
+          "type": "Number",
+          "default": 0,
+          "readAccess": "none",
+          "writeAccess": "none"
         },
-        "country": {
-          "type": "String",
-          "writeScopes": "none"
-        },
-        "iat": {
+        "lastLoginAttemptAt": {
           "type": "Date",
-          "required": true,
-          "writeScopes": "none"
+          "readAccess": "none",
+          "writeAccess": "none"
         },
-        "jti": {
-          "type": "String",
-          "required": true,
-          "writeScopes": "none"
+        "tempTokenId": {
+          "type": "String"
         },
-        "exp": {
-          "type": "Date",
-          "required": true,
-          "readScopes": "read",
-          "writeScopes": "none"
-        }
+        "authInfo": [
+          {
+            "lastUsedAt": {
+              "type": "Date",
+              "required": true
+            },
+            "userAgent": {
+              "type": "String",
+              "required": false
+            },
+            "ip": {
+              "type": "String",
+              "required": true
+            },
+            "country": {
+              "type": "String"
+            },
+            "iat": {
+              "type": "Date",
+              "required": true
+            },
+            "jti": {
+              "type": "String",
+              "required": true
+            },
+            "exp": {
+              "type": "Date",
+              "required": true
+            }
+          }
+        ]
       }
-    ],
-    "tempTokenId": {
-      "type": "String",
-      "readScopes": "none",
-      "writeScopes": "none"
     },
-    "mfaBackupCodes": [
-      {
-        "type": "String",
-        "readScopes": "none",
-        "writeScopes": "none"
-      }
-    ],
     "mfaMethod": {
       "type": "String",
       "enum": [
         "sms",
         "otp"
       ]
     },
+    "mfaBackupCodes": [
+      {
+        "type": "String"
+      }
+    ],
     "mfaSecret": {
       "type": "String",
-      "readScopes": "none",
-      "writeScopes": "none"
+      "readAccess": "none",
+      "writeAccess": "none"
     },
     "mfaPhoneNumber": {
       "type": "String",
-      "readScopes": "none",
-      "writeScopes": "none"
+      "readAccess": "none",
+      "writeAccess": "none"
     }
```

Comment:
these are not $private ?

and what about roles

Reply:
so the `$private` thing is something new just to be able to group scopes (and other field options together). Putting them under that is fine but I also thought it's equally valid to have `mfa` related fields together in the same location so that you can see them at a glance.

Btw this all looks like sudden additions here but I did make sure to document all this stuff [here](https://github.com/bedrockio/model#scopes)